### ISSUE TITLE
Event handler with zero arguments doesn't fire

### ISSUE THREAD

**Issue by wrnrlr** [solidjs/solid#1047]
### Describe the bug

There is a subtle issue when registering an event handler on an element when using hyper dom script.
When the handler has zero arguments like `onClick()` it will ever fire while this is not a problem `onClick(e)`

### Your Example Website or App

solid/solid

### Steps to Reproduce the Bug or Issue

```
function Button(props) {
  return h('button',props)
}

function App() {
  return [
    h(Button,{onClick(e){console.log('Hi')}}, "Ok"),
    h(Button,{onClick(){console.log('Hi')}}, "Fail")
  ]
}
```

### Expected behavior

I would expect their not to be a difference in behaviour

### Platform

- Solid 1.4.3

**Comment by ryansolid** [comment]
This has come up a few times and sort of stuck between rock and a hard place as we can't tell what's an event handler or a signal for prop wrapping. As I see it our options are:

1. Use the argument count as we do now. 
2. Hijack all component props that start with `on`
3. Add a special helper you need to wrap component event handers with like `onClick=event(() => ...)`

None of these are particularly great. But I think 2 is a no-go with general compatibility.

**Comment by atk** [comment]
4. Document the issue correctly (i.e. "add an unused `_` argument to your handler or wrap it in a thunk").

**Comment by ryansolid** [comment]
Yeah not sure if people are aware. I did add readme's a little while back. This could serve as a source for some documentation. Like check out the HyperScript one: https://github.com/solidjs/solid/tree/main/packages/solid/h

**Comment by atk** [comment]
Let's check if we can add JSDoc comments to these attributes.

**Comment by ryansolid** [comment]
> Let's check if we can add JSDoc comments to these attributes.

For HyperScript not sure how to do that. In fact, we don't have TS for it outside of the React-style JSX compilation. Maybe that's enough. For Tagged Template literals haven't figured out how to type them yet.
